packages/openneuro-server/src/graphql/resolvers/dataset-search.js

```diff
@@ -44,15 +45,19 @@ export const elasticRelayConnection = (
   const count = body.hits.total.value
   const lastMatch = body.hits.hits[body.hits.hits.length - 1]
   return {
-    edges: body.hits.hits
-      .map(hit => ({
-        node: childResolvers.dataset(
+    edges: body.hits.hits.map(async hit => {
+      try {
+        const node = await childResolvers.dataset(
           null,
           { id: hit._source.id },
           { user, userInfo },
-        ),
-      }))
-      .filter(edge => edge !== null), // remove datasets that user does not have permissions for
+        )
+        return { node }
```

Comment:
This can just return the dataset promise instead of wrapping it in another async function.